andy/src/main/java/nl/tudelft/cse1110/andy/grade/GradeCalculator.java

```diff
@@ -26,6 +26,16 @@ public int calculateFinalGrade(GradeValues gradeValues, GradeWeight weights) {
         if(finalGrade < 0 || finalGrade > 100)
             throw new RuntimeException("Invalid grade calculation");
 
+        if(gradeValues.getPenalty() < 0){
```

Comment:
Do we also need to check whether this number is greater than 100? Or is that not a problem?

Reply:
Martin says: That is not a problem; in fact, if there are two failing code checks and each has a penalty of 100, the total penalty will be 200, making the final grade 0.

Then, I think this deserves a code comment somewhere, and a remark in the docs!